File: connectors/connector-mantis/src/main/java/com/taskadapter/connector/mantis/MantisTaskSaver.java
package com.taskadapter.connector.mantis;

import biz.futureware.mantis.rpc.soap.client.IssueData;
import com.taskadapter.connector.common.BasicIssueSaveAPI;
import com.taskadapter.connector.definition.TaskId;
import com.taskadapter.connector.definition.exceptions.ConnectorException;
import com.taskadapter.connector.definition.exceptions.EntityProcessingException;

import java.rmi.RemoteException;

class MantisTaskSaver implements BasicIssueSaveAPI<IssueData> {

    private final MantisManager mgr;

    public MantisTaskSaver(MantisManager mgr) {
        this.mgr = mgr;
    }

    @Override
    public TaskId createTask(IssueData nativeTask) throws ConnectorException {
        try {
            return mgr.createIssue(nativeTask);
        } catch (RemoteException e) {
            throw MantisUtils.convertException(e);
        } catch (RequiredItemException e) {
            throw new EntityProcessingException(e);
        }
    }

    @Override
    public void updateTask(IssueData nativeTask) throws ConnectorException {
        try {
            mgr.updateIssue(nativeTask);
        } catch (RemoteException e) {
            throw MantisUtils.convertException(e);
        }
    }
}

File: connectors/connector-mantis/src/main/scala/com/taskadapter/connector/mantis/MantisTaskSaver.scala
package com.taskadapter.connector.mantis

import java.rmi.RemoteException

import biz.futureware.mantis.rpc.soap.client.IssueData
import com.taskadapter.connector.common.BasicIssueSaveAPI
import com.taskadapter.connector.definition.TaskId
import com.taskadapter.connector.definition.exceptions.{ConnectorException, EntityProcessingException}


final class MantisTaskSaver(val mgr: MantisManager) extends BasicIssueSaveAPI[IssueData] {
  @throws[ConnectorException]
  override def createTask(nativeTask: IssueData): TaskId = try {
    mgr.createIssue(nativeTask)
  } catch {
    case e: RemoteException =>
      throw MantisUtils.convertException(e)
    case e: RequiredItemException =>
      throw new EntityProcessingException(e)
  }

  @throws[ConnectorException]
  override def updateTask(nativeTask: IssueData): Unit = {
    try
      mgr.updateIssue(nativeTask)
    catch {
      case e: RemoteException =>
        throw MantisUtils.convertException(e)
    }
  }
}
